File: c-codes/ledBlink-1.c

/* This code demonstrates basic usage of wiringPi library */
/* It is the most basic LED blinking code (0.5 sec duty cycle) */
/* An LED is connected to pin 11 (GPIO 17) and ground */

/* To compile -  gcc -Wall -o ledBlink1 ledBlink-1.c -lwiringPi */
/* Needs sudo to run. So use sudo ./ledBlink1 */

#include <wiringPi.h>

int main(void)
{
    wiringPiSetup();		/* Initialize  the library */

    pinMode(17, OUTPUT);	/* Set GPIO17 as output */

    while (1)
    {
    	digitalWrite(0, HIGH);	/* Switch on the LED */
	delay (500);

	digitalWrite(0, LOW);  /* Switch off the LED */
	delay (500);
    }

    return 0;
}

File: python-codes/gpiozero/ledBlink-1.py
# Basic program to blink an LED using gpiozero library
# The LED is connected to pin 11 on board
# Pin 11 is GPIO 17 on the raspi in BCM mode
# Default pinmode is always BCM mode in gpioZero

from gpiozero import LED
from time import sleep


led = LED(17)	# LED anode at pin 11 (GPIO 17) of the pi

while True:
    led.on()
    sleep(1)
    led.off()
    sleep(1)

    
